File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(std::string s, int k) {
      
        std::ios_base::sync_with_stdio(false);
        std::cin.tie(NULL);

        int count = 0;
        auto isVowel = [](char ch) {
            return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
        };

       for (int i = 0; i < k; i++) {
            if (isVowel(s[i])) {
                count++;
            }
        }

        int maxCount = count;

       for (int i = k; i < s.length(); i++) {
           
            if (isVowel(s[i - k])) {
                count--;
            }
           
            if (isVowel(s[i])) {
                count++;
            }

            maxCount = std::max(maxCount, count);

           
            if (maxCount == k) return k;
        }

        return maxCount;
    }
};

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.dart
import 'dart:math';

class Solution {
  int maxVowels(String s, int k) {
    int count = 0;
    
   for (int i = 0; i < k; i++) {
      if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u') {
        count++;
      }
    }
    
   int maxCount = count;

   for (int i = k; i < s.length; i++) {
     
    if (s[i - k] == 'a' || s[i - k] == 'e' || s[i - k] == 'i' || s[i - k] == 'o' || s[i - k] == 'u') {
        count--;
      }
     
      if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u') {
        count++;
      }
      
      maxCount = max(maxCount, count);
    }
    
    return maxCount;
  }
}